ogip_spectra/ogip_spectrum_dataset.py

```diff
@@ -42,44 +43,127 @@ class StandardOGIPDataset(SpectrumDatasetOnOff):
         Name of the dataset.
 
     """
+
     stat_type = "wstat"
     tag = "StandardOGIPDataset"
 
-    def __init__(
-        self,
-        *args,
-        **kwargs
-    ):
-        axis = kwargs.pop("grouping_axis",None)
+    def __init__(self, *args, **kwargs):
+        self._grouped = None
+        axis = kwargs.pop("grouping_axis", None)
         super().__init__(*args, **kwargs)
-        self._grouped_dataset = None
         self.grouping_axis = axis
+        self.apply_grouping(self.grouping_axis)
+
+    @property
+    def grouped(self):
+        """Return grouped SpectrumDatasetOnOff."""
+        return self._grouped
 
     @property
-    def grouping_axis(self):
-        """Energy axis providing energy grouping for stat calculations."""
-        return self._grouping_axis
-
-    @grouping_axis.setter
-    def grouping_axis(self, axis):
-        """Energy axis providing energy grouping for stat calculations."""
-        self._grouping_axis = axis
-        if axis is not None:
-            self._apply_grouping()
-
-    def _apply_grouping(self):
+    def _is_grouped(self):
+        return self.grouped is not None
+
+    def apply_grouping(self, axis=None):
         """Apply grouping."""
-        self._grouped_dataset = self.resample_energy_axis(self.grouping_axis, name=f"group_{self.name}")
+        if axis is None:
+            raise ValueError(
+                "A grouping MapAxis must be provided."
+            )
+        else:
+            dataset = self.to_spectrum_dataset()
+            self._grouped = dataset.resample_energy_axis(
+                axis, name=f"group_{self.name}"
+            )
 
     @property
-    def grouped(self):
-        """Return grouped dataset."""
-        return self._grouped_dataset
+    def models(self):
+        """Models (`~gammapy.modeling.models.Models`)."""
+        return self.grouped._models
+
+    @models.setter
+    def models(self, models):
+        """Models setter"""
+        if self._is_grouped:
```

Comment:
here you probably have to set the models on `self` as well no?
Why `fset` here?

Reply:
@registerrier thank you for looking at this.

> here you probably have to set the models on self as well no?

Why is this necessary? I also override the `models` property to redirect to the `self.grouped._models`, so it should be handled correctly internally without the `self` having any model really set on it

 

> Why fset here?
Indeed, no idea. I simiplfied it to `self.grouped.models = models`

If the rest looks OK I'll add tests and push. 

A general question: in the future how do we want to integrate this in Gammapy? Add it to the main code (my first choice I guess) or leave it as a user-contributed recipe?